app/src/main/java/org/p2p/wallet/appsfly/AppsFlyerService.kt

```diff
@@ -6,26 +6,37 @@ import com.appsflyer.AppsFlyerConversionListener
 import com.appsflyer.AppsFlyerLib
 import com.google.firebase.messaging.RemoteMessage
 import org.p2p.wallet.BuildConfig
+import timber.log.Timber
+
+private const val UNINSTALL_APP_KEY = "af-uinstall-tracking"
+private const val TAG = "AppsFlyerService"
 
 class AppsFlyerService(private val context: Context) {
 
     private val listener: AppsFlyerConversionListener = AppsFlyerConversionListenerImpl()
 
     fun install(application: Application, devKey: String) {
-        val appsFlyer = AppsFlyerLib.getInstance().apply {
-            setDebugLog(BuildConfig.DEBUG)
+        val appsFlyer = AppsFlyerLib.getInstance()
+        kotlin.runCatching {
+            appsFlyer.setDebugLog(BuildConfig.DEBUG)
+            appsFlyer.init(devKey, listener, application)
+            appsFlyer.start(application)
+        }.onSuccess {
+            Timber.tag(TAG).i("AppsFlyer service is initialized")
+        }.onFailure {
+            Timber.tag(TAG).i("AppsFlyer service, error on init  = $it")
         }
-        appsFlyer.init(devKey, listener, application)
-        appsFlyer.start(application)
     }
 
     fun onNewToken(newToken: String) {
+        val appsFlyerInstance = AppsFlyerLib.getInstance()
```

Comment:
to a field i meant to do as a property 

val appsFlyerInstance 
get() = AppsFlyerLib.getInstance()